### PR TITLE
Allow private connection for Github self-hosted runner

### PR DESCRIPTION
**Pull request by tonybolzan**
Private connections in k8s Cluster is a desired security tip. And running a self-hosted runner in a private VCN can accomplish the connections without a public IP.

With this modification public and private connections are allowed without braking compatibility.

### CODE REVIEW

**Comment by Djelibeybi** [comment]
Thanks! Have you signed the Oracle Contributor Agreement? If not, please visit https://oca.opensource.oracle.com to sign it.

**Comment by Djelibeybi** [comment]
This probably needs some error checking/catching or restrictions to only be available for runners with a specific tag.

**Comment by tonybolzan** [comment]
Can be a specific configuration on `with:` ? like `allowPrivateAccess` or other better name.
What you think ?
```yml
    steps:
      - uses: oracle-actions/configure-kubectl-oke@v1
        with:
          cluster: ${{ secrets.OKE_CLUSTER_OCID }}
          allowPrivateAccess: true
```

**Comment by Djelibeybi** [comment]
That seems reasonable to me.

**Comment by tonybolzan** [comment]
> Thanks! Have you signed the Oracle Contributor Agreement? If not, please visit https://oca.opensource.oracle.com to sign it. Adding another commit to the PR will also trigger the Bot to verify that it's been signed.

The OCA has bean rejected, I submited another trial with a updated information.
But this project is not in the OCA project list, so I had to select `other` in the project list.
And I received this message `Other: For "Other" project(s), reach out to the project owner(s) to add the project(s) to the system.` on [this](https://oca.opensource.oracle.com/?ojr=contrib-create%3Btype%3Dindividual) form

**Comment by Djelibeybi** [comment]
I have added the project to the OCA application, so you should be able to submit now. Sorry for the hurdles, but you're the first person to submit a PR to any of our actions. :)

**Comment by tonybolzan** [comment]
I'm waiting for the OCA to be approved to update the PR.

**Comment by tonybolzan** [comment]
PR Updated, you can review.

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Use this GitHub Action to install and configure `kubectl` to connect to the spec
 
 ## Prerequisites
 
-The target OKE cluster must have a **public Kubernetes API Endpoint** in order for the GitHub Action to successfully connect to the cluster.
+The target OKE cluster must have a **public Kubernetes API Endpoint** or you must run on a [Self-Hosted GitHub Action][6] to enable Private Endpoint with Inputs in order for the GitHub Action to successfully connect to the cluster.
```
Comment (Djelibeybi) [inline README.md:7]:
```suggestion
The target OKE cluster must have a **public Kubernetes API Endpoint** or you must deploy a [self-hosted GitHub Runner][6] to an instance on the same private subnet to enable Private Endpoint with Inputs in order for the GitHub Action to successfully connect to the cluster.
```

File: README.md [hunk 2]
```diff
@@ -19,10 +19,11 @@ We recommend using GitHub Secrets to store these values. [Defining your environm
 ## Inputs
 
 * `cluster`: (Required) The OCID of the OKE cluster to configure
+* `enablePrivateEndpoint`: (Optional) If you want to connect on a Private Kubernetes API Endpoint. Default: false
```
Comment (Djelibeybi) [inline README.md:22]:
```suggestion
* `enablePrivateEndpoint`: (Optional) set this to 'true' if you need to connect to a private Kubernetes API endpoint. Requires a self-hosted GitHub Runner deployed to an instance on the same private subnet. Default: false
```

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

**Review by Djelibeybi — APPROVED** [review]

**Comment by Djelibeybi** [comment]
I merged this manually, so closing the PR.

**Pull request closed without merging**